_monitoring-plugins/ad/api.md

```diff
@@ -668,1043 +1929,1016 @@ POST _plugins/_anomaly_detection/detectors/results/_search
-              "feature_name": "latency_max",
-              "data": 5709
-            }
-          ],
-          "execution_end_time": 1604709684971,
-          "anomaly_grade": 0.06542056074767538,
-          "entity": [
-            {
-              "name": "host",
-              "value": "i-00f28ec1eb8997685"
-            }
-          ]
-        },
-        "sort": [
-          1604709684971
-        ]
-      },
-      {
-        "_index": ".opendistro-anomaly-results-history-2020.11.07-1",
-        "_type": "_doc",
-        "_id": "ZiIcoHUBTpMGN-4KhhVA",
-        "_version": 1,
-        "_seq_no": 79,
-        "_primary_term": 1,
-        "_score": null,
-        "_source": {
-          "detector_id": "4CIGoHUBTpMGN-4KzBQg",
-          "schema_version": 2,
-          "anomaly_score": 3.187717536855158,
-          "execution_start_time": 1604709025343,
-          "data_end_time": 1604709014520,
-          "confidence": 0.8301116064308817,
-          "data_start_time": 1604708954520,
-          "feature_data": [
-            {
-              "feature_id": "0Kld3HUBhpHMyt2e_UHn",
-              "feature_name": "latency_max",
-              "data": 441
-            }
-          ],
-          "execution_end_time": 1604709025344,
-          "anomaly_grade": 0.040767386091133916,
-          "entity": [
-            {
-              "name": "host",
-              "value": "i-00f28ec1eb8997685"
-            }
-          ]
-        },
-        "sort": [
-          1604709025344
-        ]
-      }
-    ]
+    "hits": []
+  },
+  "aggregations": {
+    "test": {
+      "doc_count_error_upper_bound": 0,
+      "sum_other_doc_count": 0,
+      "buckets": [
+        {
+          "key": "FINISHED",
+          "doc_count": 32
+        }
+      ]
+    }
   }
 }
 ```
 
-In historical detectors, specify the `detector_id`.
-To get the latest task:
+---
 
-#### Request
+## Search detector result
+Introduced 1.0
+{: .label .label-purple }
 
-```json
-GET _plugins/_anomaly_detection/detectors/<detector_id>?task=true
-```
+Returns all results for a search query.
 
-To query the anomaly results with `task_id`:
+To search anomaly results for `grade` greater than 0 for real-time analysis:
 
 #### Request
 
 ```json
 GET _plugins/_anomaly_detection/detectors/results/_search
+POST _plugins/_anomaly_detection/detectors/results/_search
 {
   "query": {
-    "term": {
-      "task_id": {
-        "value": "NnlV9HUBQxqfQ7vBJNzy"
-      }
+    "bool": {
+      "filter": [
+        {
+          "term": {
+            "detector_id": "Zi5zTXwBwf_U8gjUTfJG"
+          }
+        },
+        {
+          "range": {
+            "anomaly_grade": {
+              "gt": 0
+            }
+          }
+        }
+      ],
+      "must_not": [
+        {
+          "exists": {
+            "field": "task_id"
+          }
+        }
+      ]
     }
   }
 }
 ```
 
+Because real-time analysis doesn't have a batch task, the task ID in the anomaly result is null.
```

Comment:
Like Kaituo said, the batch task may be confusing. How about rephrase this sentence as "Real-time detection doesn't persist task ID in the anomaly result, so task id will be null."